cmake/mode.common.cmake

```diff
@@ -86,8 +86,10 @@ endfunction()
 add_compile_options("-ffile-prefix-map=${CMAKE_BINARY_DIR}=.")
 
 default_target_arch(target_arch)
+set(Scylla_TARGET_ARCH "${target_arch}" CACHE STRING
```

Comment:
It seems to be cumbersome.
It is better to remove `target_arch` completely and `set` `Scylla_TARGET_ARCH` to be `CACHE STRING PARENT_SCOPE` inside `default_target_arch()`.

Reply:
this is illegal in CMake. please refer to https://cmake.org/cmake/help/latest/command/set.html .

Reply:
```cmake
function(default_target_arch arch)
  set(x86_instruction_sets i386 i686 x86_64)
  if(CMAKE_SYSTEM_PROCESSOR IN_LIST x86_instruction_sets)
    set(${arch} "westmere" PARENT_SCOPE)
  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
    # we always use intrinsics like vmull.p64 for speeding up crc32 calculations
    # on the aarch64 architectures, and they require the crypto extension, so
    # we have to add "+crypto" in the architecture flags passed to -march. the
    # same applies to crc32 instructions, which need the ARMv8-A CRC32 extension
    # please note, Seastar also sets -march when compiled with DPDK enabled.
    set(${arch} "armv8-a+crc+crypto" CACHE STRING)
  else()
    set(${arch} "" CACHE STRING)
  endif()
  set(${arch} ${arch} PARENT_SCOPE) # <-------------------- added line
endfunction()
```

And the usage:
```cmake
default_target_arch(Scylla_TARGET_ARCH)
```

Please kindly explain why this wouldn't not work.

Reply:
`CACHE STRING PARENT_SCOPE` does not work. and i'd prefer not using `set(... PARENT_SCOPE)`.